### PR TITLE
feat: Create a release build for pushing artifacts #4

### PR DESCRIPTION
**Pull request by kgress**
Feature #4. Apart from the bullet points below, once this ticket merges against master we will see a release for version 1.0.0 proper. Moving forward, all new versions will not include the `beta` tag.

* Added a travis ci yaml for CI/CD. This pipeline supports conventional commits on merges against master. For full read up on convention commits, visit https://www.conventionalcommits.org/en/v1.0.0-beta.4/
* Added a maven settings file for the CI 
* Updated POM versions to indicate local development. With this setup, we won't be pushing versioning to github and instead only rely on the CI to version and release to maven central.
* Added the versions plugin to support versioning through CI
* Updated the documentation to provide additional insight on required files and also how to run the testing

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+env:
+  global:
+  # This environment variable is to support the rollover from the initial beta artifact release
+  - BETA_TAG="-beta"
+before_script:
+  # Before updating versioning and pushing artifacts, copy over the maven settings file with ossrh and gpg credentials
+  - cp .travis.settings.xml $HOME/.m2/settings.xml
+script:
+  # Find the most recent version via git tags
+  - OLD_VERSION=$(npx git-semver-tags | head -n 1)
```
Comment (skang0601) [inline .travis.yml]: I'm not completely sure if the image this will run on will have npx or anything like that installed.

File: .travis.yml [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+  # Using conventional commits paradigm, check the merge commit message on master to set up a new version deployment
+  - if [ -n "$OLD_VERSION" ]; then
```
Comment (skang0601) [inline .travis.yml]: Another thing to consider is to break this job into two jobs/stages and pass the version as an artifact between teh two.

File: .travis.yml (hunk 1 above)
Comment (kgress) [reply .travis.yml]:
```
$ npx --version
9.6.0

$ gpg --version
gpg (GnuPG) 1.4.16
```

NPX and GPG are both on the vm

File: .travis.yml (hunk 2 above)
Comment (kgress) [reply .travis.yml]: I like the idea of having everything in one pipeline. It makes it a little easier to manage imo.

Comment (kgress) [reply .travis.yml]: I see what you mean regarding the jobs for the pipeline. I'm experimenting with that now

**Pull request merged**